### ISSUE TITLE
Allow Curly Brace Block Delimiters for precedence

### ISSUE THREAD

**Issue by CyborgMaster** [bbatsov/rubocop#2627]
Based of the following lines in `lib/rubocop/cop/style/block_delimiters.rb`
```  ruby
          # If there are no parentheses around the arguments, then braces and
          # do-end have different meaning due to how they bind, so we allow
          # either.
```
it seems that RuboCop should allow either delimiter type ({} or do/end) if operator precedence would make a difference.  However the following code:
``` ruby
        cr.stubs client: mock {
          expects(:email_disabled=).with(true)
          expects :save
        }
```
generates this offence:
```
C: Style/BlockDelimiters: Avoid using {...} for multi-line blocks.
        cr.stubs client: mock {
```

If I was to use do/end here, the block would be passed to the `stubs` method, but I need it to be passed to the `mock` method, is there something I'm missing or doing wrong here?

**Comment by alexdowad** [comment]
This is a bug. `BlockDelimiters` just checks if the last arg is a `block` node, but in this case it's a `hash` node, whose last `pair` contains a `block` node.

**Comment by alexdowad** [comment]
Just pushed a fix to my open PR.

**Comment by CyborgMaster** [comment]
Awesome!  Thanks a ton.
